dpd/p4/sidecar.p4

```diff
@@ -2130,7 +2147,7 @@ control Egress(
 		} else if (is_mcast == true) {
 			mcast_ctr.count(eg_intr_md.egress_port);
 
-			if (is_ipv6_mcast) {
+			if (!is_ipv6_mcast) {
```

Comment:
At line 2122 we have:
```
is_ipv6_mcast = (ipv6_prefix != 16w0xff02);
```
`ff02` is for link-local multicast, so `is_ipv6_mcast` means any non-linklocal multicast, right?  So we want to bump the link-local counter for `!is_ipv6_mcast`?  Or am I getting turned around by all the `not`s?

Reply:
The logic or possibly the variable naming seems kind of confused here, before this change. `0xff` is the multi-cast prefix, and `0x02` means "link-local scope". So `is_ipv6_mcast` is true for anything that is not _exactly_ a link-local multicast address. It is true for multicast addresses of other scopes, or for addresses that aren't multicast at all like link-local addresses or globally-unique addresses.

What are we trying to catch here? _Only_ link-local multicast addresses? If that's true, then I'd probably invert the prefix check on L2122 and rename the variable to `is_ipv6_ll_mcast` or something like that.

Reply:
@Nieuwejaar yeah. My bad. I think this got lost in the review changes. This should be renamed to `is_link_local_mcastv6` and be   

```
is_link_local_ipv6_mcast = (ipv6_prefix == 16w0xff02);
```

`egress_rid` will capture replicated mcast packets. So, I think it's renamed for the counter, and we don't need the `!`.

Reply:
> The logic or possibly the variable naming seems kind of confused here, before this change. `0xff` is the multi-cast prefix, and `0x02` means "link-local scope". So `is_ipv6_mcast` is true for anything that is not _exactly_ a link-local multicast address. It is true for multicast addresses of other scopes, or for addresses that aren't multicast at all like link-local addresses or globally-unique addresses.
> 
> What are we trying to catch here? _Only_ link-local multicast addresses? If that's true, then I'd probably invert the prefix check on L2122 and rename the variable to `is_ipv6_ll_mcast` or something like that.

Yeah. What I said in my comment (before I expanded yours @bnaecker) haha.

Reply:
I made the suggested change to `is_link_local_ipv6_mcast`